### ISSUE TITLE
docker based testing workflow broken - nokogiri

### ISSUE THREAD

**Issue by mattf** [radanalyticsio/radanalyticsio.github.io#86]
```
$ docker run --rm -it -p 4000:4000 radanalytics.io 
fetch http://mirror.envygeeks.io/alpine/v3.5/main/x86_64/APKINDEX.tar.gz
fetch http://mirror.envygeeks.io/alpine/v3.5/community/x86_64/APKINDEX.tar.gz
fetch http://mirror.envygeeks.io/alpine/edge/community/x86_64/APKINDEX.tar.gz
fetch http://mirror.envygeeks.io/alpine/edge/testing/x86_64/APKINDEX.tar.gz
fetch http://mirror.envygeeks.io/alpine/edge/main/x86_64/APKINDEX.tar.gz
OK: 255 MiB in 72 packages
Fetching gem metadata from https://rubygems.org/...............
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies...
Fetching public_suffix 2.0.5 (was 1.5.3)
Using colorator 1.1.0
Fetching sass 3.4.24 (was 3.4.22)
Fetching rb-fsevent 0.10.2 (was 0.9.7)
Fetching ffi 1.9.18 (was 1.9.14)
Fetching kramdown 1.13.2 (was 1.11.1)
Using liquid 3.0.6
Using mercenary 0.3.6
Using forwardable-extended 2.6.0
Using rouge 1.11.1
Using safe_yaml 1.0.4
Fetching coffee-script-source 1.12.2 (was 1.10.0)
Fetching minitest 5.10.2 (was 5.9.1)
Fetching thread_safe 0.3.6 (was 0.3.5)
Using net-dns 0.8.0
Using execjs 2.7.0
Fetching i18n 0.8.4 (was 0.7.0)
Using multipart-post 2.0.0
Using jekyll-paginate 1.1.0
Using jekyll-swiss 0.4.0
Fetching gemoji 3.0.0 (was 2.1.0)
Fetching mini_portile2 2.2.0 (was 2.1.0)
Installing rb-fsevent 0.10.2 (was 0.9.7)
Installing public_suffix 2.0.5 (was 1.5.3)
Installing mini_portile2 2.2.0 (was 2.1.0)
Installing gemoji 3.0.0 (was 2.1.0)
Installing i18n 0.8.4 (was 0.7.0)
Installing coffee-script-source 1.12.2 (was 1.10.0)
Installing minitest 5.10.2 (was 5.9.1)
Installing thread_safe 0.3.6 (was 0.3.5)
Fetching unicode-display_width 1.3.0 (was 1.1.1)
Using bundler 1.15.1
Using pathutil 0.14.0
Fetching faraday 0.12.1 (was 0.9.2)
Using coffee-script 2.4.1
Fetching nokogiri 1.8.0 (was 1.6.8.1)
Installing kramdown 1.13.2 (was 1.11.1)
Using jekyll-coffeescript 1.0.1
Fetching addressable 2.5.1 (was 2.4.0)
Installing sass 3.4.24 (was 3.4.22)
Installing faraday 0.12.1 (was 0.9.2)
Installing unicode-display_width 1.3.0 (was 1.1.1)
Fetching terminal-table 1.8.0 (was 1.7.3)
Fetching tzinfo 1.2.3 (was 1.2.2)
Installing addressable 2.5.1 (was 2.4.0)
Installing ffi 1.9.18 (was 1.9.14) with native extensions
Fetching sawyer 0.8.1 (was 0.7.0)
Installing tzinfo 1.2.3 (was 1.2.2)
Installing terminal-table 1.8.0 (was 1.7.3)
Fetching jekyll-sass-converter 1.5.0 (was 1.3.0)
Installing sawyer 0.8.1 (was 0.7.0)
Installing jekyll-sass-converter 1.5.0 (was 1.3.0)
Fetching octokit 4.7.0 (was 4.3.0)
Installing octokit 4.7.0 (was 4.3.0)
Fetching activesupport 4.2.8 (was 4.2.7)
Using jekyll-gist 1.4.0
Installing activesupport 4.2.8 (was 4.2.7)
Installing nokogiri 1.8.0 (was 1.6.8.1) with native extensions
Fetching rb-inotify 0.9.10 (was 0.9.7)
Fetching ethon 0.10.1 (was 0.9.1)
Installing rb-inotify 0.9.10 (was 0.9.7)
Using listen 3.0.6
Using jekyll-watch 1.5.0
Fetching jekyll 3.4.3 (was 3.2.1)
Installing ethon 0.10.1 (was 0.9.1)
Using typhoeus 0.8.0
Fetching github-pages-health-check 1.3.4 (was 1.2.0)
Installing jekyll 3.4.3 (was 3.2.1)
Installing github-pages-health-check 1.3.4 (was 1.2.0)
Fetching minima 2.1.1 (was 1.2.0)
Fetching jekyll-avatar 0.4.2
Fetching jekyll-default-layout 0.1.4
Fetching jekyll-feed 0.9.2 (was 0.5.1)
Fetching jekyll-github-metadata 2.4.0 (was 2.1.1)
Fetching jekyll-optional-front-matter 0.1.2
Fetching jekyll-readme-index 0.1.0
Fetching jekyll-redirect-from 0.12.1 (was 0.11.0)
Fetching jekyll-relative-links 0.4.1
Fetching jekyll-seo-tag 2.2.3 (was 2.0.0)
Fetching jekyll-sitemap 1.0.0 (was 0.10.0)
Installing jekyll-avatar 0.4.2
Installing minima 2.1.1 (was 1.2.0)
Installing jekyll-default-layout 0.1.4
Installing jekyll-optional-front-matter 0.1.2
Installing jekyll-readme-index 0.1.0
Installing jekyll-relative-links 0.4.1
Installing jekyll-github-metadata 2.4.0 (was 2.1.1)
Installing jekyll-redirect-from 0.12.1 (was 0.11.0)
Installing jekyll-feed 0.9.2 (was 0.5.1)
Fetching jekyll-theme-architect 0.0.4
Installing jekyll-seo-tag 2.2.3 (was 2.0.0)
Fetching jekyll-theme-cayman 0.0.4
Fetching jekyll-theme-dinky 0.0.4
Fetching jekyll-theme-hacker 0.0.4
Fetching jekyll-theme-leap-day 0.0.4
Fetching jekyll-theme-merlot 0.0.4
Fetching jekyll-theme-midnight 0.0.4
Installing jekyll-sitemap 1.0.0 (was 0.10.0)
Fetching jekyll-theme-minimal 0.0.4
Fetching jekyll-theme-modernist 0.0.4
Fetching jekyll-theme-primer 0.2.1
Installing jekyll-theme-architect 0.0.4
Fetching jekyll-theme-slate 0.0.4
Installing jekyll-theme-dinky 0.0.4
Installing jekyll-theme-cayman 0.0.4
Installing jekyll-theme-hacker 0.0.4
Fetching jekyll-theme-tactile 0.0.4
Fetching jekyll-theme-time-machine 0.0.4
Installing jekyll-theme-modernist 0.0.4
Fetching jekyll-titles-from-headings 0.2.0
Installing jekyll-theme-slate 0.0.4
Installing jekyll-theme-primer 0.2.1
Installing jekyll-titles-from-headings 0.2.0
Installing jekyll-theme-time-machine 0.0.4
Installing jekyll-theme-tactile 0.0.4
Installing jekyll-theme-merlot 0.0.4
Installing jekyll-theme-midnight 0.0.4
Installing jekyll-theme-minimal 0.0.4
Installing jekyll-theme-leap-day 0.0.4
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/lib/ruby/gems/2.3.0/gems/nokogiri-1.8.0/ext/nokogiri
/usr/bin/ruby -r ./siteconf20170703-18-jb3446.rb extconf.rb
checking if the C compiler accepts ... yes
Building nokogiri using packaged libraries.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/bin/$(RUBY_BASE_NAME)
	--help
	--clean
	--use-system-libraries
/usr/lib/ruby/gems/2.3.0/gems/bundler-1.15.1/lib/bundler/rubygems_integration.rb:377:in `block (2 levels)
in replace_gem': can't activate mini_portile2 (~> 2.2.0), already activated mini_portile2-2.1.0. Make
sure all dependencies are added to Gemfile. (Gem::LoadError)
	from extconf.rb:460:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /usr/lib/ruby/gems/2.3.0/extensions/x86_64-linux/2.3.0/nokogiri-1.8.0/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /usr/lib/ruby/gems/2.3.0/gems/nokogiri-1.8.0 for inspection.
Results logged to /usr/lib/ruby/gems/2.3.0/extensions/x86_64-linux/2.3.0/nokogiri-1.8.0/gem_make.out

An error occurred while installing nokogiri (1.8.0), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.8.0'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 141, which depends on
    jekyll-mentions was resolved to 1.2.0, which depends on
      html-pipeline was resolved to 2.6.0, which depends on
        nokogiri
```

**Comment by mattf** [comment]
@elmiko fyi

**Comment by elmiko** [comment]
thanks @mattf , that nokogiri is always causing issues.... /me shakes fist

**Comment by elmiko** [comment]
i have been working on a patch to remove the github pages stuff, which coincidentally removes nokogiri.

@mattf would you mind checking out #88 to see if it works on your system?

**Comment by mattf** [comment]
that works, thanks

**Comment by elmiko** [comment]
fixed by #88
